openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx

```diff
@@ -632,13 +669,85 @@ const TagsPage = () => {
         ) : (
           <div className="full-height" data-testid="tags-container">
             {currentCategory && (
-              <div
-                className="tw-flex tw-justify-between tw-items-center"
-                data-testid="header">
-                <div
-                  className="tw-text-link tw-text-base tw-py-2"
-                  data-testid="category-name">
-                  {currentCategory.displayName ?? currentCategory.name}
+              <div className="flex justify-between" data-testid="header">
+                <div className="flex items-center">
+                  {isNameEditing ? (
+                    <Row align="middle" gutter={8}>
+                      <Col>
+                        <Input
+                          className="input-width"
+                          data-testid="tag-category-name"
+                          name="tagCategoryName"
+                          value={currentCategoryName}
+                          onChange={(e) =>
+                            setCurrentCategoryName(e.target.value)
+                          }
```

Comment:
Do not directly set the state value in the onChange callback, create a separate handler and set the state there.

Reply:
What difference it will make? eventually, it will also be a callback! if callback is ha more than one line of operation then it will make sense to create a separate function but in this case, I don't feel there is need to create an extra function and assign it! Please let me know if you think of another way!

Reply:
Make sense, but creating a separate handler would be better practice here.